Jenkinsfile

```diff
@@ -1,59 +1,100 @@
+                    --info --stacktrace --refresh-dependencies \
+                    dependencies'
             }
         }
         stage('Compile') {
-            mdk_builder.inside{
-                sh '''
-                export GRADLE_USER_HOME=$(pwd)/.gradle
-                ./gradlew -PbuildNumber=$BUILD_NUMBER -PbuildAccess=$BUILD_ACCESS -PbuildTag=$GIT_TAG -PartifactoryUrl=$ARTIFACTORY_URL -PartifactoryUsername=$ARTIFACTORY_CREDENTIALS_USR -PartifactoryPassword=$ARTIFACTORY_CREDENTIALS_PSW --gradle-user-home GRADLE_USER_HOME --continue --info --stacktrace clean assemble
-                '''
+            steps {
+                sh './gradlew \
+                    -PbuildNumber=$BUILD_NUMBER \
+                    -PbuildAccess=$BUILD_ACCESS \
+                    -PbuildTag=$TAG_NAME \
+                    -PartifactoryUrl=$ARTIFACTORY_URL \
+                    -PartifactoryUsername=$ARTIFACTORY_CREDENTIALS_USR \
+                    -PartifactoryPassword=$ARTIFACTORY_CREDENTIALS_PSW \
+                    --gradle-user-home GRADLE_USER_HOME \
+                    --continue --info --stacktrace \
+                    clean assemble'
             }
         }
         stage('Test') {
-            mdk_builder.inside{
-                sh '''
-                export GRADLE_USER_HOME=$(pwd)/.gradle
-                xvfb-run ./gradlew -PbuildNumber=$BUILD_NUMBER -PbuildAccess=$BUILD_ACCESS -PbuildTag=$GIT_TAG -PartifactoryUrl=$ARTIFACTORY_URL -PartifactoryUsername=$ARTIFACTORY_CREDENTIALS_USR -PartifactoryPassword=$ARTIFACTORY_CREDENTIALS_PSW -PtestrailHost=$TESTRAIL_HOST -PtestrailUser=$TESTRAIL_CREDENTIALS_USR -PtestrailPassword=$TESTRAIL_CREDENTIALS_PSW -PtestrailSuiteId=$TESTRAIL_SUITE_ID -PadditionalTestArguments=$ADDITIONAL_TEST_ARGUMENTS --gradle-user-home GRADLE_USER_HOME --continue --info --stacktrace check testrailPublish
-                '''
+            steps {
+                sh 'xvfb-run ./gradlew \
+                    -PbuildNumber=$BUILD_NUMBER \
+                    -PbuildAccess=$BUILD_ACCESS \
+                    -PbuildTag=$TAG_NAME \
+                    -PartifactoryUrl=$ARTIFACTORY_URL \
+                    -PartifactoryUsername=$ARTIFACTORY_CREDENTIALS_USR \
+                    -PartifactoryPassword=$ARTIFACTORY_CREDENTIALS_PSW \
+                    -PtestrailHost=$TESTRAIL_HOST \
+                    -PtestrailUser=$TESTRAIL_CREDENTIALS_USR \
+                    -PtestrailPassword=$TESTRAIL_CREDENTIALS_PSW \
+                    -PtestrailSuiteId=$TESTRAIL_SUITE_ID \
+                    -PadditionalTestArguments=$ADDITIONAL_TEST_ARGUMENTS \
+                    --gradle-user-home GRADLE_USER_HOME \
+                    --continue --info --stacktrace \
+                    check testrailPublish'
+            }
+        }
+        stage('Publish Snapshot') {
+            when { branch 'develop' }
+            steps {
+                sh './gradlew \
+                -PbuildNumber=1000 \
```

Comment:
I’m guessing this was a test value to get around Artifactory complaining about duplicate build number. Can we instead taking the Jenkins build number and add a large offset, i.e. +1000 to avoid the duplicate issue? Alternatively does build number need to be numeric? If not we can use git hash which would actually increase traceability. Or git hash concatenated with Jenkins build number.

Reply:
Whoops, yep, just test code.